iroh-qrpc-types/src/p2p.rs

```diff
@@ -0,0 +1,482 @@
+use bytes::Bytes;
+use cid::Cid;
+use derive_more::{From, TryInto};
+use libp2p::{Multiaddr, PeerId};
+use quic_rpc::{
+    message::{Msg, RpcMsg, ServerStreaming},
+    Service,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+use crate::RpcResult;
+
+pub type P2pAddr = super::addr::Addr<P2pService>;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Key(pub Bytes);
+
+// rpc Version(google.protobuf.Empty) returns (VersionResponse) {}
+#[derive(Serialize, Deserialize, Debug)]
+pub struct VersionRequest;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct VersionResponse {
+    pub version: String,
```

Comment:
Could this be done as `type VersionResponse = String` instead, or do we have to use new types always?

Reply:
For the responses - unlike for the requests - you don't have to have a newtype.

The response enum would just need to have a case for String, with a From conversion from String to the response enum.

I use this to avoid having to make a newtype struct for each unit response. I guess it is a matter of taste. What I did is to only use naked results for the unit response.